app/src/main/java/com/github/featuredetectandroid/utils/PhotoAnalyzer.kt

```diff
@@ -42,6 +44,17 @@ class PhotoAnalyzer(private val imageViewModel: GrayscaleViewModel) : ImageAnaly
             width = height.also { height = width }
         }
 
+        featureDetector?.width = width
+        featureDetector?.height = height
+        val keypoints = (
+            featureDetector
+                ?.detect(luminanceArrayToRGB(oriented)) ?: Pair(
+                emptyList(),
+                emptyList()
+            )
+            )
+            .first
+        imageViewModel.setKeypointsForOutput(keypoints)
         imageViewModel.setPicture(oriented, width, height)
```

Comment:
You probably don't need to pass sizes every time here too if you'll figure out how to retrieve them once

Reply:
fixed